autosklearn/pipeline/components/regression/gradient_boosting.py

```diff
@@ -123,34 +94,24 @@ def get_properties(dataset_properties=None):
     def get_hyperparameter_search_space(dataset_properties=None):
         cs = ConfigurationSpace()
         loss = CategoricalHyperparameter(
-            "loss", ["ls", "lad", "huber", "quantile"], default_value="ls")
+            "loss", ["least_squares"], default_value="least_squares")
         learning_rate = UniformFloatHyperparameter(
             name="learning_rate", lower=0.01, upper=1, default_value=0.1, log=True)
-        n_estimators = UniformIntegerHyperparameter(
-            "n_estimators", 50, 500, default_value=100)
+        max_iter = UniformIntegerHyperparameter(
+            "max_iter", 50, 500, default_value=100)
         max_depth = UniformIntegerHyperparameter(
             name="max_depth", lower=1, upper=10, default_value=3)
-        min_samples_split = UniformIntegerHyperparameter(
-            name="min_samples_split", lower=2, upper=20, default_value=2, log=False)
         min_samples_leaf = UniformIntegerHyperparameter(
             name="min_samples_leaf", lower=1, upper=20, default_value=1, log=False)
-        min_weight_fraction_leaf = UnParametrizedHyperparameter(
-            "min_weight_fraction_leaf", 0.)
-        subsample = UniformFloatHyperparameter(
-            name="subsample", lower=0.01, upper=1.0, default_value=1.0, log=False)
-        max_features = UniformFloatHyperparameter(
-            "max_features", 0.1, 1.0, default_value=1)
         max_leaf_nodes = UnParametrizedHyperparameter(
             name="max_leaf_nodes", value="None")
-        min_impurity_decrease = UnParametrizedHyperparameter(
-            name='min_impurity_decrease', value=0.0)
-        alpha = UniformFloatHyperparameter(
-            "alpha", lower=0.75, upper=0.99, default_value=0.9)
+        max_bins = UniformIntegerHyperparameter(
+            name="max_bins", lower=2, upper=256, default_value=256, log=False)
+        l2_regularization = UniformFloatHyperparameter(
+            name="l2_regularization", lower=0., upper=1., default_value=0., log=False)
 
-        cs.add_hyperparameters([loss, learning_rate, n_estimators, max_depth,
-                                min_samples_split, min_samples_leaf,
-                                min_weight_fraction_leaf, subsample, max_features,
-                                max_leaf_nodes, min_impurity_decrease, alpha])
+        cs.add_hyperparameters([loss, learning_rate, max_iter, max_depth, 
+                                min_samples_leaf, max_leaf_nodes, max_bins, 
+                                l2_regularization])
 
-        cs.add_condition(InCondition(alpha, loss, ['huber', 'quantile']))
         return cs
```

Comment:
The code for both appears to be almost identical. I think we should make it use a common base class to avoid code duplication.

Reply:
Just for the record, we decided to don't do this one for now.